### PR TITLE
Enable wchar_t as string_view value_type 

### PR DESCRIPTION
**Pull request by schaumb**
https://github.com/Neargye/magic_enum/issues/193

### CODE REVIEW

**Comment by schaumb** [comment]
Probably it would be nicer if I separate `test.cpp` testing parts into `tests.h`, because I used an ugly `cpp include` hack for testing.

File: include/magic_enum.hpp [hunk 1]
```diff
@@ -112,11 +112,31 @@ MAGIC_ENUM_USING_ALIAS_STRING_VIEW
...
 #endif
 
+using char_type = string_view::value_type;
+
 // If need another string type, define the macro MAGIC_ENUM_USING_ALIAS_STRING.
 #if defined(MAGIC_ENUM_USING_ALIAS_STRING)
 MAGIC_ENUM_USING_ALIAS_STRING
 #else
-using std::string;
+using string = std::basic_string<char_type>;
```
Comment (Neargye) [inline include/magic_enum.hpp]: see https://github.com/Neargye/magic_enum/issues/80

Comment (schaumb) [reply include/magic_enum.hpp]: Thanks! I solve it with reverting this modification.

Example usage in `test_wchar_t.cpp`

**Comment by JulienTheron** [comment]
I've just tested this with Visual Studio 2019 16.11.16, and there's an issue with __FUNCSIG__ macro, as it returns a char*.
I get the following compiler error:
magic_enum.hpp(393,38): error C2664: 'magic_enum::string_view magic_enum::detail::pretty_name(magic_enum::string_view) noexcept': cannot convert argument 1 from 'initializer list' to 'magic_enum::string_view'

File: include/magic_enum.hpp [hunk 2]
```diff
@@ -864,6 +900,7 @@ constexpr std::invoke_result_t<ResultGetterType> constexpr_switch(
     BinaryPredicate&& pred = {}) {
   using result_t = std::invoke_result_t<ResultGetterType>;
   using hash_t = std::conditional_t<no_duplicate<GlobValues, Hash>(), Hash, typename Hash::secondary_hash>;
+  static_assert(no_duplicate<GlobValues, hash_t>(), "duplicated hash found, please report it: https://github.com/Neargye/magic_enum/issues");
```
Comment (Neargye) [inline include/magic_enum.hpp:903]: It's fail on my local ubuntu & gcc-9

**Comment by JulienTheron** [comment]
Any news about this?

**Comment by Neargye** [comment]
Work on it will be merged next.

**Comment by Neargye** [comment]
Too many conflicts, made a separate PR https://github.com/Neargye/magic_enum/pull/272 based on this.

**Pull request closed without merging**